docs/docs/dev_docs/contracts/compiling.md

```diff
@@ -1,22 +1,136 @@
+
+```ts showLineNumbers
+export class PrivateTokenContract extends ContractBase {
+  /** Creates a contract instance at the given address. */
+  public static async at(address: AztecAddress, wallet: Wallet) { ... }
+
+  /** Creates a tx to deploy a new instance of this contract. */
+  public static deploy(rpc: AztecRPC, initial_supply: FieldLike, owner: FieldLike) { ... }
+
+  /** Type-safe wrappers for the public methods exposed by the contract. */
+  public methods!: {
+    /** getBalance(owner: field) */
+    getBalance: ((owner: FieldLike) => ContractFunctionInteraction) & Pick<ContractMethod, 'selector'>;
+
+    /** mint(amount: field, owner: field) */
+    mint: ((amount: FieldLike, owner: FieldLike) => ContractFunctionInteraction) & Pick<ContractMethod, 'selector'>;
+
+    /** transfer(amount: field, sender: field, recipient: field) */
+    transfer: ((amount: FieldLike, sender: FieldLike, recipient: FieldLike) => ContractFunctionInteraction) &
+      Pick<ContractMethod, 'selector'>;
+  };
+}
+```
+
+### Noir interfaces
+
+A Noir contract can [call a function](./functions.md) in another contract via `context.call_private_function` or `context.call_public_function`. However, this requires manually assembling the function selector and manually serialising the arguments, which is not type-safe.
+
+To make this easier, the compiler can generate a contract interface struct that exposes a convenience method for each function listed in a given contract ABI. These structs are intended to be used from another contract project that calls into the current one.
+
+To generate them, include a `--interface` option in the compile command with ap ath to the target folder for the generated Noir interface files:
+
+```
+aztec-cli compile --interface ../client/src/noir ./src/noir
+```
+
+Example code generated from the [PrivateToken](https://github.com/AztecProtocol/aztec-packages/blob/master/yarn-project/noir-contracts/src/contracts/private_token_contract/src/main.nr) contract:
+
+```rust
+impl PrivateTokenContractInterface {
+  fn at(address: Field) -> Self {
+      Self { address }
+  }
+  
+  fn mint(
+    self, context: &mut PrivateContext, amount: Field, owner: Field
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 2];
+    serialised_args[0] = amount;
+    serialised_args[1] = owner;
+
+    context.call_private_function(self.address, 0x1dc9c3c0, serialised_args)
+  }
+  
+
+  fn transfer(
+    self, context: &mut PrivateContext, amount: Field, sender: Field, recipient: Field
+  ) -> [Field; RETURN_VALUES_LENGTH] {
+    let mut serialised_args = [0; 3];
+    serialised_args[0] = amount;
+    serialised_args[1] = sender;
+    serialised_args[2] = recipient;
+
+    context.call_private_function(self.address, 0xdcd4c318, serialised_args)
+  }
+}
+
+```
+
```

Comment:
Do we want to give an example of how one would make a call to one of these interface methods, from some other Noir contract?

Reply:
Added a link to functions.md#contract-interface